File: VB/WebFormsDashboardDataSources/OrderDetail.vb
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.ComponentModel.DataAnnotations

Namespace WebFormsDashboardDataSources

	Partial Public Class OrderDetail
		<Key>
		<Column(Order := 0)>
		<DatabaseGenerated(DatabaseGeneratedOption.None)>
		Public Property OrderID() As Integer

		<Column(Order := 1)>
		<DatabaseGenerated(DatabaseGeneratedOption.None)>
		Public Property Quantity() As Short

		<Column(Order := 2, TypeName := "smallmoney")>
		Public Property UnitPrice() As Decimal

		<Column(Order := 3)>
		Public Property Discount() As Single

		<Column(Order := 4)>
		<StringLength(40)>
		Public Property ProductName() As String

		<StringLength(217)>
		Public Property Supplier() As String
	End Class
End Namespace

File: CS/WebFormsDashboardDataSources/Models/OrderDetail.cs
namespace WebFormsDashboardDataSources {
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public partial class OrderDetail
    {
        [Key]
        [Column(Order = 0)]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int OrderID { get; set; }

        [Column(Order = 1)]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public short Quantity { get; set; }

        [Column(Order = 2, TypeName = "smallmoney")]
        public decimal UnitPrice { get; set; }

        [Column(Order = 3)]
        public float Discount { get; set; }

        [Column(Order = 4)]
        [StringLength(40)]
        public string ProductName { get; set; }

        [StringLength(217)]
        public string Supplier { get; set; }
    }
}
